client/engine/store/mockstore.go

```diff
@@ -39,59 +41,90 @@ func (ms MockStore) GetChannelSecretKey() *[]byte {
 func (ms MockStore) GetObjectiveById(id protocols.ObjectiveId) (obj protocols.Objective, ok bool) {
 	// todo: locking
 	obj, ok = ms.objectives[id]
+
+	// return immediately if no such objective exists
+	if !ok {
+		return nil, ok
+	}
+
+	// populate channel data
+	if directfund.IsDirectFundObjective(id) {
+		dfo := obj.(*directfund.Objective)
+		ch, err := ms.getChannelById(dfo.C.Id)
+
+		if err != nil {
+			return nil, false
+		}
+
+		dfo.C = &ch
+
+		obj = dfo
+	} else if virtualfund.IsVirtualFundObjective(id) {
+		vfo := obj.(*virtualfund.Objective)
+
+		v, err := ms.getChannelById(vfo.V.Id)
+		if err != nil {
+			return nil, false
+		}
+		vfo.V = &channel.SingleHopVirtualChannel{Channel: v}
+
+		// todo: clean these checks
+		if vfo.ToMyLeft != nil {
```

Comment:
I think most other places we assume if `vfo.ToMyLeft != nil` then `vfo.ToMyLeft.Channel != nil` but it's probably safer to check both! Could we compress the two if statements into one?